bin/cron/summer_workshops_to_gdrive

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env ruby
+require_relative '../../lib/cdo/only_one'
+exit unless only_one_running?(__FILE__)
+
+require_relative '../../dashboard/config/environment'
+require 'cdo/google_drive'
+
+# Squash an unhelpful warning that's causing a Honeybadger error
+# see https://github.com/nahi/httpclient/issues/252#issuecomment-302427338
+class WebAgent
+  class Cookie < HTTP::Cookie
+    def domain
+      original_domain
+    end
+  end
+end
+
+#
+# Uses a Google Cloud service account to write summer workshop data for this year into
+# a spreadsheet in Google Drive (with permissions locked down to our organization) for exploration
+# by our programs team.
+#
+
+subjects = [
+  Pd::SharedWorkshopConstants::SUBJECT_SUMMER_WORKSHOP,
+  Pd::SharedWorkshopConstants::SUBJECT_CSF_101,
+  Pd::SharedWorkshopConstants::SUBJECT_CSF_201
+]
+
+workshops = []
+
+# The first row is an array of strings, one for each attribute name.
+workshops << Api::V1::Pd::WorkshopDownloadSerializer.new(Pd::Workshop.first).attributes.keys.map(&:to_s)
+
+# Subsequent rows are for each workshop.
+Pd::Workshop.where(subject: subjects).find_each.each do |w|
+  if w.workshop_starting_date && w.workshop_starting_date >= Date.new(2020, 5, 15) && w.workshop_starting_date <= Date.new(2020, 8, 31)
+    workshops << Api::V1::Pd::WorkshopDownloadSerializer.new(w).attributes.values
```

Comment:
Quick check on the cost of doing this filtering in the script instead of as part of our query:

1. We're loading all workshops ever for these three subjects, rather than only those in a very specific range.
2. Making the range part of the query would require a join against the first session and might be more likely to introduce a bug.

Given the number of rows we're working with, I suspect this is fine through August of this year.  Instincts on when we'd want to rethink this?

Reply:
Yes, I wanted to avoid the join and also adding actual SQL to do the range query, and did have a similar concern about complexity.  It currently takes about 30 seconds to do the query, which returns 4341 workshops before we filter by date.  Those workshops go back to 2016.  I think that we could cope with this query even taking several minutes, which I think will give us a few more years of creating workshops.

Reply:
Sounds good - thanks for walking me through the napkin-math on this!